install/kubernetes/cilium/templates/validate.yaml

```
@@ -45,3 +45,13 @@
     {{ fail "Hubble UI requires hubble.ui.frontend.image.tag to be '>=v0.9.0'" }}
   {{- end }}
 {{- end }}
+
+{{- if .Values.ingressController.enabled }}
+  {{- if hasKey .Values "kubeProxyReplacement" }}
+    {{- if eq .Values.kubeProxyReplacement "disabled" }}
```

nit: Since the failure message is the same either way, how about
```suggestion
  {{- if and (ne .Values.kubeProxyReplacement "partial") (ne .Values.kubeProxyReplacement "strict") }}
```